org.eclipse.ui.console/src/org/eclipse/ui/console/TextConsole.java

```diff
@@ -400,25 +395,24 @@ public IHyperlink getHyperlink(int offset) {
 	 * Binary search for the position at a given offset.
 	 *
 	 * @param offset    the offset whose position should be found
-	 * @param positions the positions list to search in
 	 * @return the position containing the offset, or <code>null</code>
 	 */
-	private Position findPosition(int offset, Position[] positions) {
+	private ConsoleHyperlinkPosition findPosition(int offset) {
 
-		if (positions.length == 0) {
+		if (fPositions.isEmpty()) {
 			return null;
 		}
 
 		int left= 0;
```

Comment:
Can't we replace all this selfmade searching in this method with a call to https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Arrays.html#binarySearch(T%5B%5D,int,int,T,java.util.Comparator) ?

Reply:
Yes done !